### PR TITLE
fix light header test

### LINKED ISSUE #917: Light Client Header Broken Test
In the `testSingleAggreatorSingleFullNodeSingleLightNode` test in [full node integration test](https://github.com/rollkit/rollkit/blob/main/node/full_node_integration_test.go#L326), the Light node is not initialized correctly.

The nodes are created with [createNode](https://github.com/rollkit/rollkit/blob/main/node/full_node_integration_test.go#L538), which calls `newFullNode`. It then sets `config.Light = true`, before starting the node. This doesn't make it a light node, just a full node with a config option.

### PR DESCRIPTION
**Pull request by S1nus**
attempt to fix light client header sync test

fixed it by specifying `InitialHeight=1` in the `GenesisDoc`. Not entirely sure why this is needed for just the light node and not the full nodes

Fixes #917

### CODE REVIEW

**Comment by nashqueue** [comment]
@S1nus Please Use the format "Fixes #917" in the pr to link them together and give it more context

**Comment by Manav-Aggarwal** [comment]
Documented the light node needing this initial height to be set to 1 here: https://github.com/rollkit/rollkit/issues/946

**Review by Manav-Aggarwal — DISMISSED** [review]
LGTM.

File: node/full_node_integration_test.go [hunk 1]
```diff
@@ -535,25 +563,27 @@ func createNode(ctx context.Context, n int, isMalicious bool, aggregator bool, d
...
 			P2P:                p2pConfig,
 			DALayer:            "mock",
 			Aggregator:         aggregator,
 			BlockManagerConfig: bmConfig,
+			Light:              isLight,
 		},
 		keys[n],
 		signingKey,
 		proxy.NewLocalClientCreator(app),
-		&tmtypes.GenesisDoc{ChainID: "test", Validators: genesisValidators},
```
Comment (gupadhyaya) [inline node/full_node_integration_test.go:549]: don't need to add additional logic for only `isLight`, but you can set `InitialHeight = 1` for all types of nodes as everything starts at height 1.

File: node/full_node_integration_test.go [hunk 2]
```diff
@@ -476,15 +494,25 @@ func createNodes(aggCtx, ctx context.Context, num int, isMalicious bool, wg *syn
...
-	nodes[0], apps[0] = createNode(aggCtx, 0, isMalicious, true, dalc, keys, wg, t)
+	node, app := createNode(aggCtx, 0, isMalicious, true, false, keys, wg, t)
+	apps[0] = app
+	nodes[0] = node.(*FullNode)
+	// use same, common DALC, so nodes can share data
+	nodes[0].dalc = dalc
+	nodes[0].blockManager.SetDALC(dalc)
 	for i := 1; i < num; i++ {
-		nodes[i], apps[i] = createNode(ctx, i, isMalicious, false, dalc, keys, wg, t)
+		node, app := createNode(ctx, i, isMalicious, false, false, keys, wg, t)
```
Comment (gupadhyaya) [inline node/full_node_integration_test.go:504]: you can use `node, apps[i]` here

File: node/full_node_integration_test.go [hunk 3]
```diff
@@ -476,15 +494,25 @@ func createNodes(aggCtx, ctx context.Context, num int, isMalicious bool, wg *syn
...
+		nodes[i] = node.(*FullNode)
+		apps[i] = app
+		nodes[i] = node.(*FullNode)
```
Comment (gupadhyaya) [inline node/full_node_integration_test.go:507]: duplicate stmt?

File: node/full_node_integration_test.go [hunk 4]
```diff
@@ -312,33 +311,52 @@ func testSingleAggreatorSingleFullNodeSingleLightNode(t *testing.T) {
...
-	time.Sleep(2 * time.Second) // wait for more than 1 blocktime for syncer to work
-	require.NoError(node2.Start())
+	sequencer.(*FullNode).dalc = dalc
+	sequencer.(*FullNode).blockManager.SetDALC(dalc)
+	fullNode.(*FullNode).dalc = dalc
+	fullNode.(*FullNode).blockManager.SetDALC(dalc)
 
-	node3.conf.Light = true
-	require.NoError(node3.Start())
+	lightNode, _ := createNode(
```
Comment (gupadhyaya) [inline node/full_node_integration_test.go:332]: [nitpick] merge all separate lines for each argument into 1 line. don't need separate lines.

File: node/full_node_integration_test.go (hunk 3 above)
Comment (Manav-Aggarwal) [reply node/full_node_integration_test.go]: removed

File: node/full_node_integration_test.go (hunk 4 above)
Comment (Manav-Aggarwal) [reply node/full_node_integration_test.go]: done

File: node/full_node_integration_test.go (hunk 2 above)
Comment (Manav-Aggarwal) [reply node/full_node_integration_test.go]: done

File: node/full_node_integration_test.go (hunk 1 above)
Comment (Manav-Aggarwal) [reply node/full_node_integration_test.go:549]: done

**Review by Manav-Aggarwal — APPROVED** [review]

**Review by gupadhyaya — APPROVED** [review]

**Pull request merged**